sdk/python/feast/feature_server.py

```diff
@@ -246,24 +257,26 @@ async def get_online_features(request: GetOnlineFeaturesRequest) -> Dict[str, An
         dependencies=[Depends(inject_user_details)],
     )
     async def retrieve_online_documents(
-        request: GetOnlineFeaturesRequest,
+        request: GetOnlineDocumentsRequest,
     ) -> Dict[str, Any]:
         logger.warning(
             "This endpoint is in alpha and will be moved to /get-online-features when stable."
         )
         # Initialize parameters for FeatureStore.retrieve_online_documents_v2(...) call
         features = await run_in_threadpool(_get_features, request, store)
 
-        read_params = dict(
-            features=features,
-            full_feature_names=request.full_feature_names,
-            query=request.query_embedding,
-            query_string=request.query_string,
-        )
+        read_params = dict(features=features, query=request.query, top_k=request.top_k)
+        if request.api_version == 2 and request.query_string is not None:
+            read_params["query_string"] = request.query_string
 
-        response = await run_in_threadpool(
-            lambda: store.retrieve_online_documents_v2(**read_params)  # type: ignore
-        )
+        if request.api_version == 2:
```

Comment:
this shouldn't be necessary. v2 vs v1 should be approximately the same where v2 was meant to incorporate more fields. If there is a difference, it's a bug.